src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -242,7 +244,51 @@ internal int SerializeLogRecord(LogRecord logRecord)
         var timestamp = logRecord.Timestamp;
         var cursor = 0;
         cursor = MessagePackSerializer.WriteArrayHeader(buffer, cursor, 3);
-        cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+
+        var categoryName = logRecord.CategoryName;
+        string eventName = null;
+
+        // If user configured explicit TableName, use it.
+        if (this.m_tableMappings != null && this.m_tableMappings.TryGetValue(categoryName, out eventName))
+        {
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+        }
+        else if (!this.shouldPassThruTableMappings)
+        {
+            eventName = this.m_defaultEventName;
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+        }
+        else if (this.shouldPassThruTableMappings)
+        {
+            if (!rawNameToSanitizedName.TryGetValue(categoryName, out var sanitizedName))
+            {
+                if (categoryName.Length > 0)
+                {
+                    int cursorStartIdx = cursor;
+                    int validNameLength = 0;
+                    cursor = Sanitize(buffer, cursor, ref cursorStartIdx, ref validNameLength, categoryName);
+                    if (validNameLength > 0)
+                    {
+                        eventName = Encoding.UTF8.GetString(buffer, cursorStartIdx + 2, validNameLength);
```

Comment:
This will cause heap alloc, and in extreme cases (e.g. we keep getting new category names and we've run out of the cache capacity), it would cause unnecessary memory pressure without benefit.

Reply:
I've used `Span<T>` and added a overloaded function for `AddPartAField`, which takes `Span<T>` as one of its argument to address the above mentioning issue. Thank you.